### PR TITLE
8265439: [TestBug] Enable and fix ignored unit tests in MenuItemTest

### PR DESCRIPTION
**Pull request by aghaisas**
This PR enables ignored unit tests from MenuItemTest and fixes them.
4 ignored tests are fixed. 
2 ignored tests are removed.

**Before fix :**
total tests = 89
failures = 0
ignored tests =  6

**After fix :**
total tests = 87
failures = 0
ignored tests =  0

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265439](https://bugs.openjdk.java.net/browse/JDK-8265439): [TestBug] Enable and fix ignored unit tests in MenuItemTest

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/466/head:pull/466` \
`$ git checkout pull/466`

Update a local copy of the PR: \
`$ git checkout pull/466` \
`$ git pull https://git.openjdk.java.net/jfx pull/466/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 466`

View PR using the GUI difftool: \
`$ git pr show -t 466`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/466.diff">https://git.openjdk.java.net/jfx/pull/466.diff</a>

</details>

### CODE REVIEW

**Comment by kevinrushforth** [comment]
It looks like you concluded that the implementation is correct, and that the ignored tests were making bad assumptions. Can you add your evaluation to this PR explaining why the tests are incorrect?

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java [hunk 1]
```diff
@@ -419,22 +406,36 @@
         assertEquals(other.get(), menuItem.isVisible());
     }
 
-    @Ignore // keyCharacter for keyCodeCombination cannot be null
-    @Test public void setSpecifiedAccelerator1() {
+    @Test(expected=NullPointerException.class)
+    public void setSpecifiedAccelerator_nullKeyCombination1() {
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:410]: Constructing KeyCodeCombination with a null KeyCode & constructing a KeyCharacterCombination with null key character results in NullPointerException as present in KeyCombinationTest.java.

These tests were ignored with a comment on similar lines. I have enabled them and added expected NullPointerException.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java [hunk 2]
```diff
@@ -453,18 +454,15 @@
...
-        Modifier[] modifierArray = {};
-        KeyCombination kc = new KeyCharacterCombination(null, modifierArray);
+        KeyCombination kc = new KeyCharacterCombination("A", KeyCombination.ALT_DOWN);
         SimpleObjectProperty<KeyCombination> other = new SimpleObjectProperty<KeyCombination>(menuItem, "accelerator", kc);
         menuItem.acceleratorProperty().bind(other);
         assertEquals(kc, menuItem.getAccelerator());
     }
 
-    @Ignore
     @Test public void getUnspecifiedMnemonicParsing() {
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:464]: Test was incorrect. I have corrected it.
Default value of mnemonicParsing property of MenuItem is true.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java [hunk 3]
```diff
@@ -453,18 +454,15 @@
         assertNotNull(menuItem.acceleratorProperty());
     }
 
-    @Ignore // keyCharacter cannot be null for keyCharacterCombination
     @Test public void acceleratorCanBeBound() {
-        Modifier[] modifierArray = {};
-        KeyCombination kc = new KeyCharacterCombination(null, modifierArray);
+        KeyCombination kc = new KeyCharacterCombination("A", KeyCombination.ALT_DOWN);
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:458]: Corrected the test case by creating a valid KeyCombination.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java [hunk 4]
```diff
@@ -275,19 +268,13 @@
...
     }
 
     @Test public void getUnspecifiedGraphicProperty2() {
         MenuItem mi2 = new MenuItem("",null);
         assertNotNull(mi2.graphicProperty());
-    }
-
-    @Ignore // Again, calling graphicPropery() is not ensuring a non null graphic
-    // node.
-    @Test public void unsetGraphicButNotNull() {
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:287]: Test was incorrect. The value of the 'graphic' property is not set and asserting on not null is incorrect.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java [hunk 5]
```diff
@@ -228,20 +227,14 @@
...
     }
 
     @Test public void getUnspecifiedTextProperty2() {
         MenuItem mi2 = new MenuItem("");
         assertEquals("", mi2.getText());
     }
 
-    @Ignore // calling textProperty will no ensure text value is non null
-    @Test public void unsetTextButNotNull() {
-        MenuItem mi2 = new MenuItem();
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:240]: Test was incorrect. The value of the 'text' property is not set and asserting on not null is incorrect.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java [hunk 6]
```diff
@@ -419,22 +406,36 @@
...
+    @Test(expected=NullPointerException.class)
+    public void setSpecifiedAccelerator_nullKeyCombination2() {
         Modifier[] modifierArray = {};
         KeyCombination kc = new KeyCharacterCombination(null, modifierArray);
         menuItem.setAccelerator(kc);
         assertEquals(kc, menuItem.getAccelerator());
     }
 
+    @Test public void setSpecifiedAccelerator1() {
+        Modifier[] modifierArray = {};
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:426]: Added a test case for valid KeyCodeCombination.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java [hunk 7]
```diff
@@ -419,22 +406,36 @@
...
+        KeyCombination kc = new KeyCodeCombination(KeyCode.A, modifierArray);
+        menuItem.setAccelerator(kc);
+        assertEquals(kc, menuItem.getAccelerator());
+    }
+
+    @Test public void setSpecifiedAccelerator2() {
```
Comment (aghaisas) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:432]: Added a test case for valid KeyCharacterCombination.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java (hunk 1 above)
Comment (kevinrushforth) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:410]: Where is the NPE thrown? I see that your change makes the test pass, but if it never gets to the `setAccelerator` call or the `assertEquals` statements maybe you could remove them?

**Review by kevinrushforth — COMMENTED** [review]
Thanks for the evaluation. I only have one remaining question on the tests that now expect NPE.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java (hunk 1 above)
Comment (aghaisas) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuItemTest.java:410]: You are right. We can remove these test cases.
The NPE is thrown from KeyCodeCombination and KeyCharacterCombination constructors and not from the method of MenuItem that is under test.

**Review by kevinrushforth — APPROVED** [review]
Looks good.

**Pull request closed without merging**